### DISCUSSION TITLE
Go built console type of application

### DISCUSSION THREAD

**Discussion by Dr-Hex** [Jacalz/rymdport#74] [Q&A] [answered]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/Jacalz/rymdport/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Description

By following the instruction:
```
go build
```
built the application with the console attached. It also has a size of 33,9 MB instead of release 18,0 MB.
How can I build a true GUI application?

### Rymdport version

3.3.2

### Operating System

Windows

### Operating System Version

Windows 10

### Go Compiler Version

1.20.2

**Comment by Jacalz** [comment, marked as answer]
This is just Windows being silly with their defaults.

In order to not build with a console attached, you need to pass `-H=windowsgui` to the ldflags when building (`go build -ldflags="-H=windowsgui"`). Release builds want to pass `-s -w` to the ldflags (`-ldflags="-s -w"`). If you want both, you can add them both to the ldflags (it is the same for all Go applications and all Windows binaries require the flag to not open a console). It might be simpler to just use the `fyne` command to package and install per https://developer.fyne.io/started/packaging.

**Reply by Dr-Hex** [reply]
Thanks.
Now, with:
```
go build -ldflags="-H=windowsgui -s -w"
```
optional header contains "Windows GUI" subsystem flag.
The size of executable is 15,5 MB. for v3.3.3 which is close to release.
The difference to release as of now, are such the resulting exe missing all the resources:
 - no icon
 - no version info
 - no manifest

Even, if above flags are common "problem" for windows build, maybe it's worth to mention all of them in the compile instruction, to follow the exact match with the release, regardless of the of knowledge level of person who try to compile it?

Meanwhile, I'll leave a link for myself explaining what does other "ldflags" mean: https://pkg.go.dev/cmd/link
```
-s - Omit the symbol table and debug information.
-w - Omit the DWARF symbol table.
```

**Reply by Jacalz** [reply]
Yeah, sorry about that. I agree with you and should have explained it more clearly. I will make sure to update the README with better compilation steps. Basically, you need to download the `fyne` command in the link that I sent and then build using `fyne build` or `fyne build -release` if you want the metadata added (the latter enables update checking on startup and adds the size reduction ldflags mentioned above).

**Reply by Jacalz** [reply]
It should hopefully be a bit clearer with https://github.com/Jacalz/rymdport/commit/27a42331b2d47c9a444842305bc3c89cd21a8649 added.

**Reply by Jacalz** [reply]
There is now also a page at https://github.com/Jacalz/rymdport/wiki/Compiling with more information.
